### ISSUE TITLE
"Invalid hosted git fragment" for branch references with '/'

### ISSUE THREAD

**Issue by ankon** [yarnpkg/yarn#855]
**Do you want to request a *feature* or report a *bug*?**
Bug.

**What is the current behavior?**
Yarn fails to install with a package.json like this:
~~~~
{
  "name": "yarn-git-ref-bug",
  "version": "1.0.0",
  "author": "Andreas Kohn <andreas.kohn@gmail.com>",
  "license": "MIT",
  "dependencies": {
    "gulp-maven-deploy": "Collaborne/gulp-maven-deploy#fix/artifactId-overrides-default"
  }
}
~~~~

~~~~
$ yarn
yarn install v0.15.1
info No lockfile found.
[1/4] Resolving packages...
error Invalid hosted git fragment "Collaborne/gulp-maven-deploy#fix/artifactId-overrides-default".
info Visit http://yarnpkg.com/en/docs/cli/install for documentation about this command.
~~~~

**What is the expected behavior?**
Dependency is installed from the 'fix/artifactId-overrides-default' branch.

**Please mention your node.js, yarn and operating system version.**
~~~~
$ node -v
v6.7.0
$ yarn -V
0.15.1
$ uname -omsr
Linux 4.7.5-200.fc24.x86_64 x86_64 GNU/Linux
~~~~

(Bonus-bug: why '-V', if everyone else in this environment picked '-v'????)

**Comment by Kovensky** [comment]
May have been fixed by https://github.com/yarnpkg/yarn/pull/813

**Comment by kittens** [comment]
Should be fixed by #813.
